bagelbids/tests/test_merge.py

```diff
@@ -133,11 +134,39 @@ def test_merge_if_demo_has_additional_subjects(bids_json, demo_json_long):
     }
     # If there are more subjects in the demo file than the BIDS dataset
     # we expect a warning that includes the subject IDs that will be stripped
-    with pytest.warns(UserWarning, match=r"99"):
+    with pytest.warns(Warning, match=r"99"):
         result = merge_json(bids_json, demo_json_long)
     assert result == target_json
 
 
 def test_merge_if_bids_has_additional_subjects(bids_json_long, demo_json):
-    with pytest.raises(NotImplementedError):
-        merge_json(bids_json_long, demo_json)
+    # If there are more subjects in the BIDS dataset than in the demographic file
```

Comment:
@surchs I've read through `merge_json` and `merge_on_subject`. Can you explain how this assert does not fail every time? I don't see the return value of `merge_on_subject` matching `target_json`

Reply:
👍 Understood after today's walk through of the test and fixtures code.